### PR TITLE
Adding new workflow for test cases report

### PR DESCRIPTION
**Pull request by shivas-antuit**
@pratap-antuit Can you please review and merge?

### CODE REVIEW

File: workflow-templates/test-cases-ci.yml [hunk 1]
```diff
@@ -0,0 +1,83 @@
+name: ESP Allocation Test Cases CI
```
Comment (pratap-antuit) [inline workflow-templates/test-cases-ci.yml:1]: Can we change the name to generic e.g. esp-npm-test-cases-ci?

File: workflow-templates/test-cases-ci.yml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        shell: bash
+      
+      # Add Comment to Github.
+      - name: Update comment with test results
+        if: ${{ always() }}
+        uses: actions/github-script@v7.0.1
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+              const issue_number = context.issue.number;
```
Comment (pratap-antuit) [inline workflow-templates/test-cases-ci.yml:74]: can you please move this part into makefile?

File: workflow-templates/test-cases-ci.yml (hunk 1 above)
Comment (shivas-antuit) [reply workflow-templates/test-cases-ci.yml:1]: I am changing this to ESP TEST CASES CI. Can you please confirm?

Comment (knxp34) [reply workflow-templates/test-cases-ci.yml:1]: You should prepend the BU that you work for in the name This is shared across the entire company. So maybe sws-esp-npm-test-cases-ci?

File: workflow-templates/test-cases-ci.yml (hunk 2 above)
Comment (shivas-antuit) [reply workflow-templates/test-cases-ci.yml]: I have removed this section and added that into Makefile

**Pull request merged**